src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -101,7 +102,8 @@ def test_ompparallel_changes_begin_string(fortran_reader):
     assert isinstance(pdir.children[3], OMPFirstprivateClause)
     priv_clause = pdir.children[2]
 
-    # Make acopy of the loop
+    # If the code inside the region changes after lowering, the next lowering
+    # will update the clauses appropriately
```

Comment:
Is this your preferred mechanism for keep clauses up-to-date? I ask because in #2157 I've introduced an 'update' mechanism to do this: changes to ChildrenList trigger each ancestor node to update (if necessary).

Reply:
I don't particularly like how this is currently implemented.

I prefer your "sending an update signal upwards to invalidate/recalculate cached values" solution. (I would prefer even more not caching/duplicating information at all, but I know it is not always simple and your solution will be useful in several places when the duplication exists.)

We can update this after your PR is on master.

Reply:
Great. Please could you add a TODO for #2157.

Reply:
Done